cobalt/media/base/metrics_provider.cc

```diff
@@ -122,5 +124,46 @@ bool MediaMetricsProvider::IsInitialized() const {
   return media_info_.has_value();
 }
 
+void MediaMetricsProvider::StartTrackingAction(WebMediaPlayerAction action) {
+  DCHECK(IsInitialized());
+  DCHECK(!IsActionCurrentlyTracked(action));
+  ScopedLock scoped_lock(mutex_);
+
+  tracked_actions_start_times_[action] = clock_->NowTicks();
+}
+
+void MediaMetricsProvider::EndTrackingAction(WebMediaPlayerAction action) {
+  DCHECK(IsInitialized());
+  DCHECK(IsActionCurrentlyTracked(action));
+  ScopedLock scoped_lock(mutex_);
+
+  auto duration = clock_->NowTicks() - tracked_actions_start_times_[action];
+  ReportActionLatencyUMA(action, duration);
+  tracked_actions_start_times_.erase(action);
+}
+
+bool MediaMetricsProvider::IsActionCurrentlyTracked(
+    WebMediaPlayerAction action) {
+  DCHECK(IsInitialized());
+  ScopedLock scoped_lock(mutex_);
+  return tracked_actions_start_times_.find(action) !=
+         tracked_actions_start_times_.end();
+}
+
+void MediaMetricsProvider::ReportActionLatencyUMA(
+    WebMediaPlayerAction action, const base::TimeDelta& action_duration) {
+  switch (action) {
+    case WebMediaPlayerAction::SEEK:
+      UMA_HISTOGRAM_TIMES("Cobalt.Media.WebMediaPlayer.Seek.Timing",
+                          action_duration);
+      break;
+    case WebMediaPlayerAction::UNKNOWN_ACTION:
+    default:
+      UMA_HISTOGRAM_TIMES("Cobalt.Media.WebMediaPlayer.Other.Timing",
+                          action_duration);
```

Comment:
What is the anticipated usage of this "Other" histogram? Feels a bit odd as it's not measuring one thing, but potentially lots of things... which makes the timing histogram itself useless outside of a signal that something besides "SEEK" is happening. If that's the intent, just make an enum histogram where each WebMediaPlayerAction value is its own bucket and increment for each?

Reply:
I didn't expect it to get any data, as new actions should be added to new histograms instead of using the Other/Unknown action.

As such, I've removed the histogram, and let the Unknown action not report any data.